src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -23,6 +23,7 @@ case class NullConst extends ConstExpr
 case class FuncExpr(function: String, args: List[Expr], isAggregation: Boolean) extends Expr
 case class BinaryOpExpr(lhs: Expr, op: String, rhs: Expr) extends Expr
 case class TypecastExpr(lhs: Expr, rhs: String) extends Expr
+case class Placeholder extends Expr
```

Comment:
This design looks very clear. I was wondering why we need a separate `VarPatt`?

My only concern is we will separate head atom and body atom, functions taking atoms as arguments, functions dealing with expression and patterns. That may seem a bit repetitive. I might be wrong, the functions would be like the following?
```scala
def doSomethingWithExpr(e: Expr) = ...
def doSomethingWithPattern(p: Pattern) = p match {
  case ExprPatt(e) => doSomethingWithExpr(e)
  ...
}
```